ompi/mpi/fortran/mpif-h/ialltoallw_f.c

```diff
@@ -116,13 +116,10 @@ void ompi_ialltoallw_f(char *sendbuf, MPI_Fint *sendcounts,
                             c_recvtypes, c_comm, &c_request);
     if (NULL != ierr) *ierr = OMPI_INT_2_FINT(c_ierr);
     if (MPI_SUCCESS == c_ierr) *request = PMPI_Request_c2f(c_request);
+    c_request->req_flags |= OMPI_REQ_NB_RELEASE_DATA;
 
     OMPI_ARRAY_FINT_2_INT_CLEANUP(sendcounts);
```

Comment:
For a definitive fix, I am afraid we also have to handle this.

This is a no-op if `sizeof(MPI_Fint) == sizeof(int)`, but a `malloc()` under the hood otherwise (e.g. `ifort -i8 ...`).
In the latter case,

Reply:
Right, we need this for a generic solution. Right now the libnbc generates the entire plan of communications during _init, so the displacement and count arrays are unnecessary once we have the request. But this behavior is particular to libnbc, other frameworks could use the arrays of count and displacements to dynamically create communications.

Reply:
The new code is slightly better at handling all temporary arrays (displacements, counts and datatypes). However, I don't think this is the right place to save these array, instead they should be saved by the nbc module itself. The only thing we need to set at this level is the flags to indicate which one should be released and which should be left alone.